### ISSUE TITLE
[3.16.0 rc2] failure to upload files makes phone unusable

### ISSUE THREAD

**Issue by jonim8or** [nextcloud/android#8269]
### Steps to reproduce (happened to me once now... but does not always happen)
1. select photos to upload from photos app
2. click share -> nextcloud
3. Create new folder and select that
3. one file got uploaded, the rest failed to upload

### Expected behaviour
- The photos I selected should all be uploaded to the folder

### Actual behaviour
- One photo got uploaded, the rest gave annoying uppopping errors that popup over other apps, making the phone unusable. Setting phone to airplane mode temporarily solves the issue... but also makes the phone unusable. Several seconds after switching off airplane mode the popups start again

![Screenshot_20210412-094049](https://user-images.githubusercontent.com/12826624/114359942-c5160b80-9b74-11eb-855e-d3aeb376c327.png)

### Report 
************ CAUSE OF ERROR ************

java.lang.UnsupportedOperationException
at java.util.AbstractList.set(AbstractList.java:132)
at java.util.AbstractList$ListItr.set(AbstractList.java:426)
at java.util.Collections.sort(Collections.java:247)
at com.nextcloud.common.IPV6PreferringDNS.lookup(IPV6PreferringDNS.java:30)
at
okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.kt:164)
at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.kt:129)
at okhttp3.internal.connection.RouteSelector.next(RouteSelector.kt:71)
at
okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:205)
at
okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106)
at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74)
at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255)
at
okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
at
okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
at
okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
at
okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
at
okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
at
okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
at
okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
at
okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
at okhttp3.internal.connection.RealCall.execute(RealCall.kt:154)
at com.nextcloud.common.OkHttpMethodBase.execute(OkHttpMethodBase.kt:171)
at
com.nextcloud.client.network.ConnectivityServiceImpl.isInternetWalled(ConnectivityServiceImpl.java:75)
at
com.owncloud.android.utils.FilesSyncHelper.lambda$restartJobsIfNeeded$0(FilesSyncHelper.java:239)
at
com.owncloud.android.utils.-$$Lambda$FilesSyncHelper$78U1WN_nzr5HiGIrRZMThSEamz8.run(lambda)
at java.lang.Thread.run(Thread.java:761)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30160052
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Fairphone
Device: FP2
Model: FP2
Id: 19.11.2
Product: FP2

************ FIRMWARE ************
SDK: 25
Release: 7.1.2
Incremental: gms-24df0be9

**Comment by cweiske** [comment]
I have the same problem with the dev version:
```
************ APP INFORMATION ************
ID: com.nextcloud.android.beta
Version: 20210402
Build flavor: versionDev

************ DEVICE INFORMATION ************
Brand: google
Device: flo
Model: Nexus 7
Id: NJH47F
Product: razor

************ FIRMWARE ************
SDK: 25
Release: 7.1.2
Incremental: 323e49e081
```

**Comment by cweiske** [comment]
This seems to be the same problem as reported in #8067, which has been marked duplicate of #7400.

**Comment by CL-Jeremy** [comment]
Same. I don't see why that issue is a duplicate, either. I also have Nougat (7.0, Samsung Galaxy S7). A foundless thought: the culprit could very well be one of the dependencies.

**Comment by jonim8or** [comment]
I've read through the duplicates that were mention, this could indeed be a duplicate. However there are two issues here, that maybe should be handled separately:

## Failure makes phone unusable

The way the app responds to the failure, is very serious, as it makes the phone unusable. Only technical minds come up with the idea to stop the popups by switching to airplane mode. I think these questions should be raised:
- Is it helpful to immediately try again? other options include trying again after longer and longer intervals, or asking the user if he wants to try again or cancel
- Is this notification so important that it should popup over other apps? or is it fine to just show the notification in the nextcloud app itself?

## Ipv6 usage

 In my case the problem is probably that my home internet (where I run my self-hosted nextcloud) only gets an ipv4 address. (my internet provider is still in the process of switching to ipv6) When I visit https://whatismyipaddress.com/ it only shows an ipv4 address. Therefore my self-hosted nextcloud only gets an ipv4 address, and there's nothing I can do about it (except maybe switching internet provider). 
Could nextcloud client default to ipv6 but fallback to ipv4 if ipv6 fails?
Could nextcloud client have an option to configure it to only use ipv4?

I've uninstalled and reinstalled the app, trying to solve the problem, but now I can't even connect any more, probably also due to ipv6

**Comment by tobiasKaminsky** [comment]
This seems to crash as okhttp returns via Kotlin a list which is not sortable.
Can you test this https://cloud.nextcloud.com/s/2DHnFFMycBq5wnG
You can install it in parallel.
